Sources/Containerization/ContainerManager.swift

```diff
@@ -0,0 +1,165 @@
+//===----------------------------------------------------------------------===//
+// Copyright © 2025 Apple Inc. and the Containerization project authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//===----------------------------------------------------------------------===//
+
+#if os(macOS)
+
+import ContainerizationError
+import ContainerizationOCI
+import ContainerizationOS
+import Foundation
+
+import Crypto
+
+extension String {
+    fileprivate func hash() throws -> String {
+        guard let data = self.data(using: .utf8) else {
+            fatalError("\(self) could not be converted to Data")
```

Comment:
Same here. This is marked as throws, and isn't public where we'd prefer the convenience of maybe not wanting to do a `try whatever()` so can we just throw some err